cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ServiceUtils.java

```diff
@@ -234,17 +236,18 @@ public static ServicePort createServicePort(String name, int port, int targetPor
      * @param name          Name of the port
      * @param port          The port on the service which can be accessed by clients
      * @param targetPort    The port on the container / Pod where the connections will be routed
+     * @param targetPortName the name of the target port on the container / Pod where the connections will be routed
      * @param nodePort      The desired node port number
      * @param protocol      Protocol used by this port
      *
      * @return  Created port
      */
-    public static ServicePort createServicePort(String name, int port, int targetPort, Integer nodePort, String protocol)   {
+    public static ServicePort createServicePort(String name, int port, int targetPort, String targetPortName, Integer nodePort, String protocol)   {
         return new ServicePortBuilder()
                 .withName(name)
                 .withProtocol(protocol)
                 .withPort(port)
-                .withNewTargetPort(targetPort)
+                .withNewTargetPort(Objects.requireNonNullElse(targetPortName, targetPort))
```

Comment:
I think we should have another test just for the `targetPort` path - so when the `targetPortName` will be `null`. From the changes I see that only `targetPortName` is checked.